Client/Frontend/Browser/User Scripts/UserScriptType.swift

```diff
@@ -8,64 +8,80 @@ import WebKit
-        switch self {
-        case .farblingProtection:
-            return .farblingProtection
-        case .domainUserScript(let domainUserScript):
-            switch domainUserScript {
-            case .youtubeAdBlock:
-                return .youtubeAdBlock
-            case .archive:
-                return .archive
-            case .braveSearchHelper:
-                return .braveSearchHelper
-            case .braveTalkHelper:
-                return .braveTalkHelper
-            }
-        case .nacl:
-            return .nacl
-        }
+  /// Return a source typ for this script type
+  var sourceType: ScriptSourceType {
+    switch self {
+    case .farblingProtection:
+      return .farblingProtection
+    case .deAMP:
+      return .deAMP
+    case .domainUserScript(let domainUserScript):
+      switch domainUserScript {
+      case .youtubeAdBlock:
+        return .youtubeAdBlock
+      case .archive:
+        return .archive
+      case .braveSearchHelper:
+        return .braveSearchHelper
+      case .braveTalkHelper:
+        return .braveTalkHelper
+      }
+    case .nacl:
+      return .nacl
     }
+  }
 
-    /// The order in which we want to inject the scripts
-    var order: Int {
-        switch self {
-        case .nacl: return 0
-        case .farblingProtection: return 1
-        case .domainUserScript: return 2
-        }
+  /// The order in which we want to inject the scripts
+  var order: Int {
+    switch self {
+    case .nacl: return 0
+    case .deAMP: return 1
+    case .farblingProtection: return 2
+    case .domainUserScript: return 3
     }
+  }
 
-    var injectionTime: WKUserScriptInjectionTime {
-        switch self {
-        case .farblingProtection, .domainUserScript, .nacl:
-            return .atDocumentStart
-        }
+  var injectionTime: WKUserScriptInjectionTime {
+    switch self {
+    case .farblingProtection, .domainUserScript, .nacl, .deAMP:
+      return .atDocumentStart
     }
+  }
 
-    var forMainFrameOnly: Bool {
-        switch self {
-        case .farblingProtection, .domainUserScript, .nacl:
-            return false
-        }
+  var forMainFrameOnly: Bool {
+    switch self {
+    case .farblingProtection, .domainUserScript, .nacl:
+      return false
+    case .deAMP:
+      return true
     }
+  }
 
-    var contentWorld: WKContentWorld {
-        switch self {
-        case .farblingProtection, .domainUserScript:
-            return .page
-        case .nacl:
-            return .page
-        }
+  var contentWorld: WKContentWorld {
+    switch self {
+    case .farblingProtection, .domainUserScript, .nacl:
+      return .page
+    case .deAMP:
+      // Note: this has the same content world as the `cosmeticFiltersScript`
+      // executed in the `WKNavigationDelegate`. This is currently no strictly necessary
+      // but if we ever share properties amongst them, they need to be in the
+      // same content world
+      return .cosmeticFiltersSandbox
```

Comment:
Should probably have its own sandbox or `defaultClient` as I don't see any reason why it would share the sandbox of cosmetic filters.